### PR TITLE
Extract the mock api url from code to pipeline env

### PR DESCRIPTION
**Pull request by Yuki-YuXin**
### Summary:
Please go to the PR for further workhttps://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2387
This PR is closed because of the messy state during test.

### CODE REVIEW

File: azure-pipelines/pull-request-validation/common4j.yml [hunk 1]
```diff
@@ -1,8 +1,12 @@
 # File: azure-pipelines\pull-request-validation\common4j.yml
 # Description: Run test in common4j
+# Variable: 'MOCK_API_URL' was defined in the Variables tab
 # Uses Java 11
 name: $(date:yyyyMMdd)$(rev:.r)
 
+variables:
```
Comment (SammyO) [inline azure-pipelines/pull-request-validation/common4j.yml:7]: For my own understanding, why is this needed?

**Comment by SammyO** [comment]
@Yuki-YuXin I'd like to see this PR in action, i.e. run the tests using the new env var. Could you take a look at that?

**Comment by SammyO** [comment]
@Yuki-YuXin there are more instances of the mock API URL in the code base, that haven't been replaced as part of this PR. Some of them might be in MSAL. Could you take a look at that?

File: azure-pipelines/pull-request-validation/common4j.yml (hunk 1 above)
Comment (Yuki-YuXin) [reply azure-pipelines/pull-request-validation/common4j.yml:7]: https://identitydivision.visualstudio.com/Engineering/_library?itemType=VariableGroups
We can create the variable groups here.

**Comment by Yuki-YuXin** [comment]
Summarize the discussion with Silviu in terms of this work item:

IOS don’t run the tests in the yml file. They run them locally, have a stash that we apply and then discard. As in they don’t run as part of the pipeline. They have different suite for unit which run in the pipeline and integration with mock api which run only locally.

**Comment by SammyO** [comment]
> Summarize the discussion with Silviu in terms of this work item:
> 
> IOS don’t run the tests in the yml file. They run them locally, have a stash that we apply and then discard. As in they don’t run as part of the pipeline. They have different suite for unit which run in the pipeline and integration with mock api which run only locally.

What's your point here @Yuki-YuXin ?

File: common4j/src/testFixtures/java/com/microsoft/identity/common/nativeauth/ApiConstants.kt [hunk 2]
```diff
@@ -30,8 +30,8 @@ import java.net.URL
  */
 interface ApiConstants {
     companion object {
-        const val BASEPATH = "https://native-auth-mock-api.azurewebsites.net/"
-        private const val BASE_REQUEST_PATH = BASEPATH + "1234/"
+        val BASEPATH: String? by lazy { System.getenv("MOCK_API_URL") }
```
Comment (SammyO) [inline common4j/src/testFixtures/java/com/microsoft/identity/common/nativeauth/ApiConstants.kt]: This is not how we should be dealing with env vars. They should be set as part of the gradle script, in BuildConfig vars. E.g. look at how the Labs API secret is set in various build.gradle files: `buildConfigField("String", "LAB_CLIENT_SECRET", "\"$labClientSecret\"")`.

Comment (SammyO) [reply common4j/src/testFixtures/java/com/microsoft/identity/common/nativeauth/ApiConstants.kt]: Let me know if you need some help with this.

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthOAuth2Configuration.kt [hunk 3]
```diff
@@ -51,7 +50,7 @@ class NativeAuthOAuth2Configuration(
 
     companion object {
         //Base url for the mock API to make Native Auth calls. See the swagger at
-        // https://native-auth-mock-api.azurewebsites.net/doc#/ for all possible urls
+        // $(MOCK_API_URL)/doc#/ for all possible urls
         private const val MOCK_API_URL_WITH_NATIVE_AUTH_TENANT = "https://native-auth-mock-api.azurewebsites.net/lumonconvergedps.onmicrosoft.com"
```
Comment (SammyO) [inline common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthOAuth2Configuration.kt]: Mock API value is still set here. Can you please do a ctrl-F in the entire project for this API value string and remove all occurrences?

File: azure-pipelines/pull-request-validation/common4j.yml [hunk 4]
```diff
@@ -17,6 +21,8 @@ pool:
 jobs:
 - job: build_test
   displayName: Build & Test
+  env:
+    MOCK_API_URL: https://native-auth-mock-api.azurewebsites.net/
```
Comment (SammyO) [inline azure-pipelines/pull-request-validation/common4j.yml]: Why is this here? Shouldn't this PR be introducing the integration with the environment var, so we can test that?

**Comment by Yuki-YuXin** [comment]
> > Summarize the discussion with Silviu in terms of this work item:
> > IOS don’t run the tests in the yml file. They run them locally, have a stash that we apply and then discard. As in they don’t run as part of the pipeline. They have different suite for unit which run in the pipeline and integration with mock api which run only locally.
> 
> What's your point here @Yuki-YuXin ?

They don't seem to intend to integrate the mock api into the pipeline in the center server and they can easily accept the changes of mock api because every time they use the url locally.

**Comment by Yuki-YuXin** [comment]
> @Yuki-YuXin I'd like to see this PR in action, i.e. run the tests using the new env var. Could you take a look at that?

From the experience of cert PR, if we can pass the PR pipeline verification check, we prove that the new env var actually work

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthOAuth2Configuration.kt [hunk 5]
```diff
@@ -51,8 +50,9 @@ class NativeAuthOAuth2Configuration(
 
     companion object {
         //Base url for the mock API to make Native Auth calls. See the swagger at
-        // https://native-auth-mock-api.azurewebsites.net/doc#/ for all possible urls
-        private const val MOCK_API_URL_WITH_NATIVE_AUTH_TENANT = "https://native-auth-mock-api.azurewebsites.net/lumonconvergedps.onmicrosoft.com"
+        // $(MOCK_API_URL)/doc#/ for all possible urls
+        private val BASE_PATH = BuildValues.getMockApiUrlForNativeAuthAuthority()
```
Comment (Yuki-YuXin) [inline common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthOAuth2Configuration.kt:54]: The same here. Frist use the concrete string to pass the pipeline since it's not a test file.

File: common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthOAuth2Configuration.kt (hunk 3 above)
Comment (Yuki-YuXin) [reply common4j/src/main/com/microsoft/identity/common/java/nativeauth/providers/NativeAuthOAuth2Configuration.kt]: I want to first let the PR passed first using the concrete string since it's not a test file.
